fread fails when reading mostly empty file

```
>>> text = "Nothing" + "\n" * 1000000
>>> dt.fread(text)
  3% |█▉                                                | Reading data [error]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError: Assertion 'c.get_start() >= input_start && c.get_end() <= input_end' failed in src/core/read/parallel_reader.cc, line 139
```

_Originally posted by:_ @pseudotensor

--- a/docs/releases/v1.1.0.rst
+++ b/docs/releases/v1.1.0.rst
@@ -31,6 +31,9 @@
         ``'-'``, ``'+'``, or ``'.'``. Previously, these values were allowed but
         they produced errors during parsing. [#3065]
 
+    -[bug] :func:`fread()` will no longer fail while reading mostly empty
+        files. [#3055]
+
 
     General
     -------

--- a/src/core/read/fread/fread_parallel_reader.cc
+++ b/src/core/read/fread/fread_parallel_reader.cc
@@ -54,7 +54,7 @@ void FreadParallelReader::adjust_chunk_coordinates(
   if (cc.is_end_approximate()) {
     const char* end = cc.get_end();
     while (*end=='\n' || *end=='\r') end++;
-    cc.set_end_approximate(end + 1);
+    cc.set_end_approximate(end);
   }
 }
 

--- a/tests/fread/test-fread-issues.py
+++ b/tests/fread/test-fread-issues.py
@@ -585,3 +585,11 @@ def test_issue2943():
     assert_equals(DT[:, 'A':'C'], EXP)
     DT = dt.fread(src2)
     assert_equals(DT[:, 'A':'C'], EXP)
+
+
+def test_issue_3055():
+    repeat = 1000000
+    text = "Nothing" + "\n" * repeat
+    EXP = dt.Frame(Nothing=[None]*(repeat-1))
+    RES = dt.fread(text)
+    assert_equals(EXP,RES)

